database/tracer.go

```diff
@@ -99,4 +106,111 @@ func (t ethDBTracer) AssertPersistedGenesisBlock(genesis core.Genesis) {
 			"persisted_balance", persistedBalance.String(),
 		)
 	}
+}
+
+func (t EthDBTracer) AssertPostTxSimulationState(from common.Address, tx *types.Transaction) {
```

Comment:
[quiz] Did you get time to investigate the possibility to "fake" the current time, in order to produce predicted block header hash? I think there is a clear value in looking for alternatives for that as it would reduce the impl of the method and increase the coverage over the block hashing feature

Reply:
good idea, didn't yet, i will write a note for this